ping/rust/src/main.rs

```diff
@@ -11,8 +11,8 @@ use testground::network_conf::{
 };
 
 pub mod libp2p {
-    #[cfg(all(feature = "libp2pv0480",))]
-    pub use libp2pv0480::*;
+    #[cfg(all(feature = "libp2pmaster",))]
+    pub use libp2pmaster::*;
 
     #[cfg(all(feature = "libp2pv0470",))]
     pub use libp2pv0470::*;
```

Comment:
(creating here to get a thread)

@mxinden, I'm trying to catch other cases where the build might break after a push to `master`.

1. If `rust-libp2p` remove a feature flag in master, I guess the `libmaster` line in `Cargo.toml`will become invalid, and it will break the build. Correct?
2. `rust-libp2p` publishes an API breaking change in `master`, could it break the build of previous versions? I think the decoration `#[cfg(all(feature = "libp2pmaster",))]` completely removes the code.

Reply:
> If `rust-libp2p` remove a feature flag in master, I guess the `libmaster` line in `Cargo.toml`will become invalid, and it will break the build. Even for old versions, Correct?

Correct. Though this will change soon where we can replace all the features with a single `full`. See https://github.com/libp2p/rust-libp2p/pull/2918.

Reply:
> if `rust-libp2p` publishes an API breaking change in `master`, could it break the build of previous versions? I assume the decoration `#[cfg(all(feature = "libp2pmaster",))]` completely removes the code and we're safe here.

I am not sure I follow. Once a crate is published to crates.io it can **never** be changed again.

Reply:
Apologies, I mixed different concepts of versions,

I'm wondering if there are other cases where a push to rust-libp2p master branch can break the build for previous versions of the ping test (`cargo build --feature libp2pv0470`, for example).

- We fixed one case with this PR (when we push a version bump),
- We know another case (when we change feature flags),
- Do you see other cases?

Trying to play around with macro it looks like it's be enough, but I might miss some rust's edge cases.

```rust
// cargo build --features libp2pmaster (FAIL)
// cargo build --features libp2pv2 (OK)
fn main() {
    println!("got: {}", s() + 1)
}

#[cfg(all(feature = "libp2pmaster",))]
fn s() -> String {
    return String::from("new master API that breaks the current helloworld test");
}

#[cfg(all(feature = "libp2pv2",))]
fn s() -> i32 {
    return 2;
}
```